### PR TITLE
Issue 309 - Adding transaction isolation and default to read committed

### PR DESCRIPTION
**Pull request by aOelschlager**
For issue #309. This adds an environment variable to the mariadb image that allows you to set the transaction isolation setting. It defaults to READ-COMMITTED now.

Related issues: https://github.com/Islandora/documentation/issues/1481 and https://github.com/Islandora-Devops/islandora-playbook/issues/250

### CODE REVIEW

File: mariadb/Dockerfile [hunk 1]
```diff
@@ -23,7 +23,10 @@ RUN --mount=type=cache,id=mariadb-apk-${TARGETARCH},sharing=locked,target=/var/c
 # base image. Set to 10 minutes just incase it ran on very old or overallocated
 # hardware.
 ENV S6_CMD_WAIT_FOR_SERVICES_MAXTIME=600000
-# Default Mariadb value of 16 MB (bytes)
-ENV MYSQL_MAX_ALLOWED_PACKET=16777216
+
+ENV \ 
+    # Default Mariadb value of 16 MB (bytes)
+    MYSQL_MAX_ALLOWED_PACKET=16777216 \
+    TRANSACTION_ISOLATION=READ-COMMITTED
```
Comment (nigelgbanks) [inline mariadb/Dockerfile:30]: Please prefix `TRANSACTION_ISOLATION` with `MYSQL` to be consistent with the environmental naming conventions across the repository.

So `MYSQL_TRANSACTION_ISOLATION`.

File: mariadb/README.md [hunk 2]
```diff
@@ -41,7 +41,8 @@ default database connection configuration.
 | :------------------- | :------ | :------------------------------------------------------------------------------------ |
 | MYSQL_ROOT_PASSWORD  |         | The database root user password. Defaults to `DB_ROOT_PASSWORD`                       |
 | MYSQL_ROOT_USER      |         | The database root user (used to create the site database). Defaults to `DB_ROOT_USER` |
-| MYSQL_MAX_ALLOWED_PACKET | 16777216 | Max packet length to send to or receive from the server, [documentation](https://mariadb.com/docs/server/ref/mdb/system-variables/max_allowed_packet/) |
+| MYSQL_MAX_ALLOWED_PACKET | 16777216 | Max packet length to send to or receive from the server, [documentation](https://mariadb.com/docs/server/ref/mdb/system-variables/max_allowed_packet/) 
+| TRANSACTION_ISOLATION | READ-COMMITTED | The isolation level for transactions.
```
Comment (nigelgbanks) [inline mariadb/README.md:45]: Same prefix note as above, and please link to the documentation.

https://mariadb.com/docs/server/ref/mdb/system-variables/transaction_isolation/

File: mariadb/rootfs/etc/confd/templates/mariadb-server.cnf.tmpl [hunk 3]
```diff
@@ -9,6 +9,7 @@
 [mysqld]
 # skip-networking
 max_allowed_packet ={{ getenv "MYSQL_MAX_ALLOWED_PACKET" }}
+transaction-isolation ={{ getenv "TRANSACTION_ISOLATION" }}
```
Comment (nigelgbanks) [inline mariadb/rootfs/etc/confd/templates/mariadb-server.cnf.tmpl:12]: Same prefix note as above.

**Review by nigelgbanks — CHANGES_REQUESTED** [review]
Very good, just a few small things and it's ready to go in.

**Comment by aOelschlager** [comment]
Thank you @nigelgbanks for the feedback and for reviewing it. That makes sense to me and I made the changes that you recommended.

**Pull request merged**